EIPS/eip-x.md

```diff
@@ -0,0 +1,140 @@
+| Blocks regularly under-full, user willing to wait for a good price. | User bids below the low end of the recently accepted prices, may need to wait for a while. If waiting too long, user may need to re-submit with a higher price. | User chooses their price, and waits for it, or manually re-submits.                          | User chooses their lowest price, but also their highest price and maximum wait time, so no resubmission is needed. |
+
+In all cases, the escalator algorithm as I have described is able to perform optimally.
+
+The current gas auction model works well under half-full and less conditions, but for users with urgent needs, has the downside of overpayment. For users seeking a low price, the current model has the downside of requiring re-submission, but has the benefit of always giving users a path towards reliable block inclusion.
+
+EIP-1559 also performs well under normal conditions, but under conditions where blocks are regularly full, the price discovery mechanism breaks, and miners will fall back to the `TIP` parameter to choose the transactions to include, meaning that under network congestion, EIP-1559 forces users to _either_ choose efficient prices or certainty of next-block inclusion.
+
+EIP-1559 also has all the re-submission issues of the current model in situations where a user would like to pay under the current market rate, but has certain time constraints limiting their patience. The Escalator algorithm is the only strategy listed here that allows users to discover the lowest possible price given the network conditions and their time constraints.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+**Client-Wide Parameters**
+* `INITIAL_FORK_BLKNUM`: TBD
+
+**Transaction Parameters**
+The transaction `gasPrice` parameter is now optional, and if excluded can be replaced by these parameters instead:
+
+* `START_PRICE`: The lowest price that the user would like to pay for the transaction.
+* `START_BLOCK`: The first block that this transaction is valid at.
+* `MAX_PRICE`: The maximum price the sender would be willing to pay to have this transaction processed.
+* `MAX_BLOCK`: The last block that the user is willing to wait for the transaction to be processed in.
+
+**Proposal**
+
+For all blocks where `block.number >= INITIAL_FORK_BLKNUM`:
+
+When processing a transaction with the new pricing parameters, miners now receive a fee based off of the following linear function, where `BLOCK` is the current block number:
+
+* IF `BLOCK > MAX_BLOCK` then `TX_FEE = MAX_PRICE`.
```

Comment:
The ideal behavior imo is that the transaction cannot be included after the `MAX_BLOCK`

Reply:
Hmm, this wording conflicts with the wording above on `MAX_BLOCK`.  I disagree with @moodysalem here and think that the ideal behavior (if we have to choose between two options) is to have it sit at `MAX_PRICE` forever once `MAX_BLOCK` is reached.

Reply:
I see people complain about transactions pending for days, blocking new things they want to do, in basically every dapp's support chat. Cancelling transactions via resubmitting with different gas price is a horrible experience, and has non-deterministic results. If you really want to wait up to 3 days for a transaction, set the MAX_BLOCK to last 3 days.

Reply:
The problem is that the fee algorithm is a function of `MAX_BLOCK, START_BLOCK`, so if you set `MAX_BLOCK` to 3 days in the future, your fee increase rate is *really slow*.  I would not be opposed to adding yet-another-variable for `TERMINATION_BLOCK`, so you can escalate up to a target fee over say 5 minutes, but then sit in the pending queue for up to 24 hours at that fee.

In other words, I agree that there is a problem with never-expiring transactions.  I just disagree that `MAX_BLOCK` in this specification is the right place to fix that.  In fact, one could create a totally separate EIP that just adds in a `NOT_AFTER` field to transactions.